app/actions/views/channel.js

```diff
@@ -65,39 +55,44 @@ export function loadChannelsByTeamName(teamName, errorHandler) {
     };
 }
 
+export function getChannelSinceValue(state, channelId, postIds) {
+    const lastGetPosts = state.views.channel.lastGetPosts[channelId];
+    const lastConnectAt = state.websocket?.lastConnectAt || 0;
+
+    let since;
+    if (lastGetPosts && lastGetPosts < lastConnectAt) {
+        // Since the websocket disconnected, we may have missed some posts since then
+        since = lastGetPosts;
+    } else {
+        // Trust that we've received all posts since the last time the websocket disconnected
+        // so set `since` to the `create_at` of latest one we've received
+        const {posts} = state.entities.posts;
+        const channelPosts = postIds.map((id) => posts[id]);
+        since = getLastCreateAt(channelPosts);
+    }
+
+    return since;
+}
+
 export function loadPostsIfNecessaryWithRetry(channelId) {
     return async (dispatch, getState) => {
         const state = getState();
-        const {posts} = state.entities.posts;
-        const postsIds = getPostIdsInChannel(state, channelId);
+        const postIds = getPostIdsInChannel(state, channelId);
         const actions = [];
 
         const time = Date.now();
 
         let loadMorePostsVisible = true;
         let postAction;
-        if (!postsIds || postsIds.length < ViewTypes.POST_VISIBILITY_CHUNK_SIZE) {
+        if (!postIds || postIds.length < ViewTypes.POST_VISIBILITY_CHUNK_SIZE) {
             // Get the first page of posts if it appears we haven't gotten it yet, like the webapp
             postAction = getPosts(channelId);
         } else {
-            const lastConnectAt = state.websocket?.lastConnectAt || 0;
-            const lastGetPosts = state.views.channel.lastGetPosts[channelId];
-
-            let since;
-            if (lastGetPosts && lastGetPosts < lastConnectAt) {
-                // Since the websocket disconnected, we may have missed some posts since then
-                since = lastGetPosts;
-            } else {
-                // Trust that we've received all posts since the last time the websocket disconnected
-                // so just get any that have changed since the latest one we've received
-                const postsForChannel = postsIds.map((id) => posts[id]);
-                since = getLastCreateAt(postsForChannel);
-            }
-
+            const since = getChannelSinceValue(state, channelId, postIds);
             postAction = getPostsSince(channelId, since);
         }
 
-        const received = await retryGetPostsAction(postAction, dispatch, getState);
+        const received = await retryGetPostsAction(postAction, dispatch);
```

Comment:
maybe instead of passing `dispatch` as an arg we can change this to be `await dispatch(retryGetPostAction(postAction))` also maybe change the name of the function to be `fetchPostsActionWithRetry` or something like that